qutebrowser/browser/shared.py

```diff
@@ -339,3 +343,37 @@ def netrc_authentication(url, authenticator):
     authenticator.setPassword(password)
 
     return True
+
+
+def choose_file(multiple: bool) -> List[str]:
+    """Select file(s) for uploading, using external command defined in config.
+
+    Args:
+        multiple: Should selecting multiple files be allowed.
+
+    Return:
+        A list of selected file paths, or empty list if no file is selected.
+        If multiple is False, the return value will have at most 1 item.
+    """
+    encoding = sys.getfilesystemencoding()
+    with tempfile.NamedTemporaryFile(mode='w+', encoding=encoding) as tmpfile:
+        if multiple:
+            command = config.val.fileselect.multiple_files.command
+        else:
+            command = config.val.fileselect.single_file.command
+
+        proc = guiprocess.GUIProcess(what='choose-file')
+        proc.start(command[0],
+                   [arg.replace('{}', tmpfile.name) for arg in command[1:]])
+
+        loop = qtutils.EventLoop()
+        proc.finished.connect(lambda _code, _status: loop.exit())
+        loop.exec_()
```

Comment:
This should be `.exec()`, see 99017c50a.

Reply:
Changed